### PR TITLE
Add API docs for Caching and WebEncoders

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Part of #27538.

### CODE REVIEW

File: src/Caching/StackExchangeRedis/src/RedisCache.cs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
 namespace Microsoft.Extensions.Caching.StackExchangeRedis
 {
+
```
Comment (JunTaoLuo) [inline src/Caching/StackExchangeRedis/src/RedisCache.cs:14]: Oops, missed a commit

File: src/Caching/StackExchangeRedis/src/RedisCache.cs [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 namespace Microsoft.Extensions.Caching.StackExchangeRedis
 {
+    /// <summary>
+    /// Distributed cache implementation using Microsoft StackExchange Redis.
```
Comment (pranavkm) [inline src/Caching/StackExchangeRedis/src/RedisCache.cs:15]:
```suggestion
    /// Distributed cache implementation using Redis.
    /// <para>Uses <c>StackExchange.Redis</c> as the Redis client.</para>
```

File: src/WebEncoders/src/Testing/HtmlTestEncoder.cs [hunk 3]
```diff
@@ -12,11 +12,13 @@ namespace Microsoft.Extensions.WebEncoders.Testing
     /// </summary>
     public sealed class HtmlTestEncoder : HtmlEncoder
     {
+        /// <inheritdoc />
```
Comment (pranavkm) [inline src/WebEncoders/src/Testing/HtmlTestEncoder.cs:15]: Could we enhance the summary for this type to say something about not using it in production apps?

Comment (pranavkm) [reply src/WebEncoders/src/Testing/HtmlTestEncoder.cs:15]: Same for the other test encoders

Comment (JunTaoLuo) [reply src/WebEncoders/src/Testing/HtmlTestEncoder.cs:15]: The type summary already indicates that it's used for testing only. Didi you want to add it to all the method docs as well?

Comment (pranavkm) [reply src/WebEncoders/src/Testing/HtmlTestEncoder.cs:15]: It feels a bit spare. How about 

> `<see cref="HtmlEncoder" /> used for unit testing. This encoder does not perform any encoding and should not be used in application code.

Comment (JunTaoLuo) [reply src/WebEncoders/src/Testing/HtmlTestEncoder.cs:15]: Updated for all three encoders.

**Comment by JunTaoLuo** [comment]
ping @pranavkm for review.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**